### PR TITLE
Feature/PromiseKit Documentation

### PR DESCRIPTION
**Pull request by tonikocjan**
Rewritten documentation.

### CODE REVIEW

File: PovioKit/Classes/Utilities/PromiseKit/Combine.swift [hunk 1]
```diff
@@ -26,9 +26,7 @@ public func combine<T>(
         case .success:
           barrier.async(flags: .barrier) {
             if promises.allSatisfy({ $0.isFulfilled }) {
-              dispatchQueue.async {
-                seal.resolve(with: promises.compactMap { $0.value })
-              }
+              seal.resolve(with: promises.compactMap { $0.value }, on: dispatchQueue)
```
Comment (nejcvivod) [inline PovioKit/Classes/Utilities/PromiseKit/Combine.swift]: Could we make these changes in a separate PR, since this is titled documentation?

Comment (tonikocjan) [reply PovioKit/Classes/Utilities/PromiseKit/Combine.swift]: Done.
